### ISSUE TITLE
Validator for File Field

### ISSUE THREAD

**Issue by shivam091** [horprogs/Just-validate#32]
Previously I was using another validation plugin in my application but due to it's dependency on JQuery, I was supposed to remove it from the application.

In that plugin I was having something like below for validating file field

```javascript
file: {
  extension: "jpe,jpg,jpeg,png",
  type: "image/jpeg,image/png",
  minSize: 2 * 10 * 1024,
  maxSize: 2 * 1024 * 1024,
  message: "error message"
}
```

How can we achive something like this to validate file max, min size, extensions, etc. in Just-Validate ?

**Comment by horprogs** [comment]
It's a good point. For now Just-validate doesn't have file validation, but it makes sense to add this. I'll look into that

**Comment by horprogs** [comment]
Implemented files validation rules in `3.1.0`. You could check example here: https://just-validate.dev/examples/#files.
So now there are 3 new rules: `minFilesCount`, `maxFilesCount` (if you want to validate count of uploaded files), and `files` rule. Probably it will need some adjustments, but I guess it covers the basic needs.

For example:

```
validation
                .addField('#example7_minmax', [
                  {
                    rule: 'minFilesCount',
                    value: 1,
                  },
                  {
                    rule: 'maxFilesCount',
                    value: 3,
                  },
                ])
                .addField('#example7_png', [
                  {
                    rule: 'minFilesCount',
                    value: 1,
                  },
                  {
                    rule: 'maxFilesCount',
                    value: 1,
                  },
                  {
                    rule: 'files',
                    value: {
                      files: {
                        types: ['image/png'],
                        extensions: ['png'],
                      },
                    },
                  },
                ])
                .addField('#example7_attr', [
                  {
                    rule: 'files',
                    value: {
                      files: {
                        extensions: ['jpeg', 'jpg', 'png'],
                        maxSize: 20000,
                        minSize: 10000,
                        types: ['image/jpeg', 'image/jpg', 'image/png'],
                      },
                    },
                  },
                ])
```

**Comment by shivam091** [comment]
Great!!. The validation rules you added are really useful and covered with all basic needs of file validation. Once again great feature and thank you.
